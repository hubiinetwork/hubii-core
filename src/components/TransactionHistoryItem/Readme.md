
```js
const address = '0xb2682160c482eB985EC9F3e364eEc0a904C44C23';
<div style={{ backgroundColor: 'grey', padding: 10 }}>
  <TransactionHistoryItem
    data={{
      address: `${address}`,
      time: new Date('January 09, 1995 05:19:09'),
      amount: '0.000000000000000001',
      txnId:
        '0x4891ee9bc872f5ea35b1dd3b7384bdc4a4c26f63ee7036f83568c8612603ed63',
      to: '60c482eB985EC9F3e364eEc0a904C44C23',
      from: '60c482eB985EC9F3e364eEc0a904C44C23',
      fiatValue: '987.365',
      coin: 'UKG',
      confirmations: 204
    }}
    key={
      '0xb2682160c482eB985EC9F3e364eEc0a904C44C2360c482eB985EC9F3e364eEc0a904C44C23'
    }
  />
</div>;
```

```js
const address = '0xb2682160c482eB985EC9F3e364eEc0a904C44C23';
<div style={{ backgroundColor: 'grey', padding: 10 }}>
  <TransactionHistoryItem
    data={{
      address: `${address}`,
      time: new Date('February 17, 1995 23:24:00'),
      amount: '0.0011',
      fiatValue: '987.365',
      txnId:
        '0x4891ee9bc872f5ea35b1dd3b7384bdc4a4c26f63ee7036f83568c8612603ed63',
      to: '0xb2682160c482eB985EC9F3e364eEc0a904C44C23',
      from: '60c482eB985EC9F3e364eEc0a904C44C23',
      coin: 'VEE',
      confirmations: 204
    }}
    key={
      '0xb2682160c482eB985EC9F3e364eEc0a904C44C2360c482eB985EC9F3e364eEc0a904C44C23'
    }
  />
</div>;
```

```js
<div style={{ backgroundColor: 'grey', padding: 10 }}>
  <TransactionHistoryItem
    data={{
      time: new Date('October 23, 1995 17:24:00'),
      amount: '0.011',
      fiatValue: '987.365',
      toCoin: 'ADX',
      fromCoin: 'AIR',
      confirmations: 204
    }}
    key={
      '0xb2682160c482eB985EC9F3e364eEc0a904C44C2360c482eB985EC9F3e364eEc0a904C44C23'
    }
  />
</div>
```
