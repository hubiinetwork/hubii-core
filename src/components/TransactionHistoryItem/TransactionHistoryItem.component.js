import React from 'react';
import {
  TransactionHistoryItemCard,
  TransactionHistoryItemDate,
  Wrapper,
  Image
} from './TransactionHistoryItem.style';
import PropTypes from 'prop-types';
import TransactionHistoryDetail from '../TransactionHistoryDetail';
/**
 * This component shows history of a past transaction's detail.
 */
export const TransactionHistoryItem = props => {
  const type = !props.data.address
    ? 'exchange'
    : props.data.address === props.data.to
      ? 'received'
      : 'sent';
  return (
    <Wrapper>
      <TransactionHistoryItemDate>{props.data.date}</TransactionHistoryItemDate>
      <TransactionHistoryItemCard>
        <Image
          src={`../../../public/asset_images/${
            type === 'exchange' ? props.data.fromCoin : props.data.coin
          }.svg`}
        />

        <Image
          src={`../../../public/asset_images/${props.data.toCoin}.svg`}
          style={{
            display: type === 'received' || type === 'sent' ? 'none' : 'block'
          }}
        />
        <TransactionHistoryDetail
          type={type}
          address={type === 'received' ? props.data.from : props.data.to}
          txnId={props.data.txnId}
          amount={props.data.amount}
          usd={props.price}
          toCoin={props.data.toCoin}
          fromCoin={props.data.fromCoin}
          status={props.data.status}
          time={props.data.time}
        />
      </TransactionHistoryItemCard>
    </Wrapper>
  );
};
TransactionHistoryItem.propTypes = {
  /**
   * prop  of data of TransactionHistoryItem.
   */
  data: PropTypes.shape({
    date: PropTypes.string.isRequired,
    /**
     * Do not  pass address if you want to show exchange state of the component.
     * 'address' prop is used to pass the address of the walletHolder where 'to' and 'from' props are used to check  whether the transaction was a receving or a sending one.
     */
    address: PropTypes.string,
    time: PropTypes.string.isRequired,
    amount: PropTypes.number.isRequired,
    txnId: PropTypes.string,
    to: PropTypes.string,
    from: PropTypes.string,
    toCoin: PropTypes.string,
    fromCoin: PropTypes.string,
    coin: PropTypes.string,
    status: PropTypes.number
  }).isRequired,
  /**
   * price of 1ETH or any other currency, in dollars.
   */
  price: PropTypes.number.isRequired
};
