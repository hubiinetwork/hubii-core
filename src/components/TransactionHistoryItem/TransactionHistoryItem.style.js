import styled from 'styled-components';

export const TransactionHistoryItemCard = styled.div`
  background-color: ${({ theme }) => theme.palette.primary4};
  padding: 5px;
  align-items: center;
  border-radius: 20px;
  flex: 1;
  display: flex;
  box-shadow: ${({ theme }) => theme.shadows.light};
`;

export const Received = styled.div`
  color: ${({ theme }) => theme.palette.info} !important;
  font-family: 'SF Text';
`;

export const Sent = styled.div`
  color: ${({ theme }) => theme.palette.warning} !important;
  font-family: 'SF Text';
`;

export const Image = styled.img`
  display: flex;
  align-self: flex-start;
  align-items: center;
  height: 26px;
  width: 26px;
  margin-right: 9px;
`;

export const Wrapper = styled.div`
  display: flex;
  margin-bottom: 16px;
  align-items: center;
`;
