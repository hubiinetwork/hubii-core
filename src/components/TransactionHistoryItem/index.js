export { TransactionHistoryItem } from './TransactionHistoryItem.component';
