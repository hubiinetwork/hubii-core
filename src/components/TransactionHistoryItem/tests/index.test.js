import React from 'react';
import { shallow } from 'enzyme';

import TransactionHistoryItem from '../index';

describe('<TransactionHistoryItem />', () => {
  const props = {
    data: {
      address: '0x0000',
      time: new Date('January 09, 1995 05:19:09'),
      amount: '0.000000000000000001',
      txnId:
        '0x4891ee9bc872f5ea35b1dd3b7384bdc4a4c26f63ee7036f83568c8612603ed63',
      from: '60c482eB985EC9F3e364eEc0a904C44C23',
      fiatValue: '987.365',
      coin: 'UKG',
      confirmations: 204,
    },
  };

  it('should render correctly tx type recieved', () => {
    const wrapper = shallow(
      <TransactionHistoryItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly tx type sent', () => {
    const dataSend = { ...(props.data), from: null, to: '0x000' };
    const wrapper = shallow(
      <TransactionHistoryItem
        {...props}
        data={dataSend}
      />);
    expect(wrapper).toMatchSnapshot();
  });
});
