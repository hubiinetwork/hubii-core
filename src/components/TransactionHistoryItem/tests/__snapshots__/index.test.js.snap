// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransactionHistoryItem /> should render correctly tx type recieved 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <./index
    data={
        Object {
            "address": "0x0000",
            "amount": "0.000000000000000001",
            "coin": "UKG",
            "confirmations": 204,
            "fiatValue": "987.365",
            "from": "60c482eB985EC9F3e364eEc0a904C44C23",
            "time": 1995-01-09T04:19:09.000Z,
            "txnId": "0x4891ee9bc872f5ea35b1dd3b7384bdc4a4c26f63ee7036f83568c8612603ed63",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "data": Object {
        "address": "0x0000",
        "amount": "0.000000000000000001",
        "coin": "UKG",
        "confirmations": 204,
        "fiatValue": "987.365",
        "from": "60c482eB985EC9F3e364eEc0a904C44C23",
        "time": 1995-01-09T04:19:09.000Z,
        "txnId": "0x4891ee9bc872f5ea35b1dd3b7384bdc4a4c26f63ee7036f83568c8612603ed63",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "./index",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "data": Object {
          "address": "0x0000",
          "amount": "0.000000000000000001",
          "coin": "UKG",
          "confirmations": 204,
          "fiatValue": "987.365",
          "from": "60c482eB985EC9F3e364eEc0a904C44C23",
          "time": 1995-01-09T04:19:09.000Z,
          "txnId": "0x4891ee9bc872f5ea35b1dd3b7384bdc4a4c26f63ee7036f83568c8612603ed63",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "./index",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<TransactionHistoryItem /> should render correctly tx type sent 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <./index
    data={
        Object {
            "address": "0x0000",
            "amount": "0.000000000000000001",
            "coin": "UKG",
            "confirmations": 204,
            "fiatValue": "987.365",
            "from": null,
            "time": 1995-01-09T04:19:09.000Z,
            "to": "0x000",
            "txnId": "0x4891ee9bc872f5ea35b1dd3b7384bdc4a4c26f63ee7036f83568c8612603ed63",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "data": Object {
        "address": "0x0000",
        "amount": "0.000000000000000001",
        "coin": "UKG",
        "confirmations": 204,
        "fiatValue": "987.365",
        "from": null,
        "time": 1995-01-09T04:19:09.000Z,
        "to": "0x000",
        "txnId": "0x4891ee9bc872f5ea35b1dd3b7384bdc4a4c26f63ee7036f83568c8612603ed63",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "./index",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "data": Object {
          "address": "0x0000",
          "amount": "0.000000000000000001",
          "coin": "UKG",
          "confirmations": 204,
          "fiatValue": "987.365",
          "from": null,
          "time": 1995-01-09T04:19:09.000Z,
          "to": "0x000",
          "txnId": "0x4891ee9bc872f5ea35b1dd3b7384bdc4a4c26f63ee7036f83568c8612603ed63",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "./index",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
